pfrl/experiments/train_hrl_agent.py

```diff
@@ -0,0 +1,222 @@
+import logging
+import os
+from pfrl.agents.hrl.hiro_agent import HIROAgent
+
+from pfrl.experiments.evaluator import Evaluator
+from pfrl.experiments.evaluator import save_agent
+
+
+import os 
+import numpy as np
+
+
+def train_hrl_agent(
+    agent: HIROAgent,
+    env,
+    steps,
+    outdir,
+    checkpoint_freq=None,
+    max_episode_len=None,
+    step_offset=0,
+    evaluator=None,
+    successful_score=None,
+    step_hooks=(),
+    logger=None,
+    explore=True,
+    start_training_steps=100,
+    subgoal=None
+):
+
+    logger = logger or logging.getLogger(__name__)
+
+    episode_r = 0
+    episode_idx = 0
+
+    obs_dict = env.reset()
+    subgoal = subgoal
+
+    fg = obs_dict['desired_goal']
+    obs = obs_dict['observation']
+
+    # sample from subgoal
+    sg = subgoal.sample()
+
+    t = step_offset
+    if hasattr(agent, "t"):
+        agent.t = step_offset
+
+    episode_len = 0
+    try:
+        while t < steps:
+            # get action
+            if explore:
+                if t < start_training_steps:
+                    action = env.action_space.sample()
+                else:
+                    action = agent.act_low_level(obs, sg)
+            else:
+                action = agent.act_low_level(obs, sg)
```

Comment:
we can have `eval_mode` and `env.action_sapce` within HiroAgent so that it will be consistent with `train_agent.py`. Then we don't need this script, but use `train_agent.py`

Reply:
I can make those changes, but I think it might be better to use `train_hrl_agent.py` - there are a lot of differences between it and the standard train_agent that make it more clear to use `train_hrl_agent.py`

Reply:
I will definitely make the change to eval mode though.

Reply:
> I can make those changes, but I think it might be better to use `train_hrl_agent.py` - there are a lot of differences between it and the standard train_agent that make it more clear to use `train_hrl_agent.py`

As long as we can include subspace into env and hold env's `action_space` within HIROAgent, we can definitely use `train_agent.py`, but let's first get some good result. Meanwhile I can take care.